### PR TITLE
[Semi-modular] PCV Mk. 2

### PR DESCRIPTION
**Pull request by Stalkeros2**
## About The Pull Request
Adds PCV Mk. 2 to Black Mesa: a unique'ish, weaker variation of HEV available to player-controlled HECU. Older vests became NPC grunts' loot drop. Also gives HECU NVGs because this addition made their helmets lose the built-in flashlights.

## How This Contributes To The Skyrat Roleplay Experience
Vanguard gets more loot, and HECU players feel themselves better while being gunned down. Besides, finalising content good - I was going to add it in for a very long time.

## Changelog

:cl: Stalkeros
add: Adds PCV, a unique Black Mesa HECU-exclusive vest, available at your local Gateway. 
/:cl:

### CODE REVIEW

**Comment by Stalkeros2** [comment]
GitHub kinda died, so I can't post screenshots.

File: modular_skyrat/modules/hev_suit/code/pcv_suit.dm [hunk 1]
```diff
@@ -0,0 +1,701 @@
...
+	icon_state = "hecu_vest"
+	armor = list(MELEE = 30, BULLET = 30, LASER = 30, ENERGY = 30, BOMB = 30, BIO = 20, FIRE = 20, ACID = 20, WOUND = 10) //This is muhreens suit, of course it's mildly strong and balanced for PvP.
+	flags_inv = null
+	allowed = list(/obj/item/gun, /obj/item/ammo_box,/obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/melee/energy/sword, /obj/item/restraints/handcuffs, /obj/item/tank/internals)
+	cell = /obj/item/stock_parts/cell/hyper
+	actions_types = list(/datum/action/item_action/pcv_toggle, /datum/action/item_action/pcv_toggle_notifs)
+	resistance_flags = FIRE_PROOF|ACID_PROOF|FREEZE_PROOF
+	clothing_flags = SNUG_FIT
+
+	var/activated = FALSE
```
Comment (Zonespace27) [inline modular_skyrat/modules/hev_suit/code/pcv_suit.dm]: code doc comment these please

File: modular_skyrat/modules/hev_suit/code/pcv_suit.dm [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+	button_icon = 'modular_skyrat/modules/hev_suit/icons/toggles.dmi'
+	background_icon_state = "bg_hl"
+	icon_icon = 'modular_skyrat/modules/hev_suit/icons/toggles.dmi'
+	button_icon_state = "sound_VOICE_AND_TEXT"
+
+/datum/action/item_action/pcv_toggle_notifs/Trigger(trigger_flags)
+	var/obj/item/clothing/suit/space/pcv_suit/my_suit = target
+	var/new_setting = tgui_input_list(my_suit.current_user, "Please select your notification settings.", "PCV Notification Settings", PCV_NOTIFICATIONS)
+
+	if(!new_setting)
```
Comment (Zonespace27) [inline modular_skyrat/modules/hev_suit/code/pcv_suit.dm]:
```suggestion
	new_setting ||= PCV_NOTIFICATION_TEXT_AND_VOICE
```

File: modular_skyrat/modules/hev_suit/code/pcv_suit.dm [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+		new_setting = PCV_NOTIFICATION_TEXT_AND_VOICE
```
Comment (Zonespace27) [inline modular_skyrat/modules/hev_suit/code/pcv_suit.dm]:
```suggestion
```

File: modular_skyrat/modules/hev_suit/code/pcv_suit.dm [hunk 4]
```diff
@@ -0,0 +1,701 @@
...
+		return
+
+	if(playing_voice_line)
+		if(priority) //Shit's fucked, we better say this ASAP
+			queued_voice_lines.Insert(1, sound_in)
+		else
+			queued_voice_lines += sound_in
+		return
+
+	if(queued_voice_lines.len)
```
Comment (Zonespace27) [inline modular_skyrat/modules/hev_suit/code/pcv_suit.dm]:
```suggestion
	if(length(queued_voice_lines))
```

File: modular_skyrat/modules/hev_suit/code/pcv_suit.dm [hunk 5]
```diff
@@ -0,0 +1,701 @@
...
+			if(COMPONENT_NO_CELL)
+				failure_reason = "NO CELL INSERTED"
+			if(COMPONENT_NO_CHARGE)
+				failure_reason = "NO CELL CHARGE"
+			else
+				failure_reason = "GENERIC FAILURE"
+		send_message("ERROR - POWER SYSTEMS FAILURE - [failure_reason]", PCV_COLOR_RED)
+		return FALSE
+
+	if(!current_user.head && !istype(current_user.head, /obj/item/clothing/head/helmet/space/pcv_suit))
```
Comment (Zonespace27) [inline modular_skyrat/modules/hev_suit/code/pcv_suit.dm]: Intended to be &&, not ||?

File: modular_skyrat/modules/hev_suit/code/pcv_suit.dm [hunk 6]
```diff
@@ -0,0 +1,701 @@
...
+	var/user_old_fireloss
+	var/user_old_toxloss
+	var/user_old_cloneloss
+	var/user_old_oxyloss
+
+	var/radio_channel = RADIO_CHANNEL_COMMON
+
+	var/timer_id = null
+
+	var/voice_current_cooldown
```
Comment (Zonespace27) [inline modular_skyrat/modules/hev_suit/code/pcv_suit.dm]: don't use world.time cooldowns, use the TG `DECLARE_COOLDOWN` code instead

**Review by Zonespace27 — COMMENTED** [review]
did a quick review

File: modular_skyrat/modules/hev_suit/code/pcv_suit.dm (hunk 6 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/hev_suit/code/pcv_suit.dm]: No clue what this is, especially considering this is *the* HEV code copypate.

**Comment by Stalkeros2** [comment]
Sitting here waiting for better sprites to arrive, so I can replace everything in one go.
It's, like, completely ready code- and codersprite-wise, yet it lacks any real sprites I've showcased in Discord.
Feel free to merge rn too - I doubt that those sprites will be done, like, literally now.

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**